Sources/SwiftSDKGenerator/SystemUtils/ByteBuffer+Utils.swift

```diff
@@ -10,16 +10,23 @@
 //
 //===----------------------------------------------------------------------===//
 
+import AsyncProcess
 import Foundation
 import NIOCore
 import NIOFoundationCompat
 
 public extension ByteBuffer {
-  func unzip(isVerbose: Bool) throws -> AsyncThrowingStream<Data, any Error> {
-    let gzip = try Shell("gzip -cd", shouldLogCommands: isVerbose)
-    gzip.stdin.write(Data(buffer: self))
-    try gzip.stdin.close()
+  func unzip(isVerbose: Bool) async throws -> ByteBuffer? {
+    let result = try await ProcessExecutor.runCollectingOutput(
+      executable: "/usr/bin/gzip", ["-cd"],
+      standardInput: [self].async,
+      collectStandardOutput: true,
+      collectStandardError: false,
+      perStreamCollectionLimitBytes: 100 * 1024 * 1024
```

Comment:
that seems a bit much, that's 100 MiB, no?

Reply:
Would 10 MiB be a better choice? It's also not entirely clear if that's the top limit for collected bytes altogether or only for the intermediate buffer? What happens when the collected amount exceeds the limit?

Reply:
It's the per stream collection limit. As in: How many bytes will we collect for stdout / stderr (they're both allowed to reach the limit) before stopping and producing an error.

10 MiB isn't necessarily better but 100 MiB is quite a high limit. Usually if you need significantly more than 1 MiB or you should be streaming the data rather than ballooning it into memory.